### PR TITLE
Fix issue 3861 : XMP Dialog, Add new Groups dialog, Append Library dialog to javafx

### PR DESCRIPTION
**Pull request by AyachiNene**
Tools -> Writing XMP Data to Entry 
Groups -> Add new Group
Import->Append Library Dialog

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Thanks for your contribution! 
You should merge the latest upstream/master from jabref/master, so that github will show only show your commit and not all others which were already merged. Just add JabRef's orginal repo as remote: [Github has a help on that ](https://help.github.com/articles/fork-a-repo/#step-3-configure-git-to-sync-your-fork-with-the-original-spoon-knife-repository)

**Comment by tobiasdiez** [comment]
I went over everything and think this is now ok so (it's not perfect, but at least workable...).

File: src/main/java/org/jabref/gui/actions/WriteXMPAction.java [hunk 1]
```diff
@@ -116,49 +110,49 @@ private void writeXMP() {
                                     .map(Optional::get)
                                     .collect(Collectors.toList());
 
-            SwingUtilities.invokeLater(() -> optionsDialog.getProgressArea()
-                                                          .append(entry.getCiteKeyOptional().orElse(Localization.lang("undefined")) + "\n"));
+            Platform.runLater(() -> optionsDialog.getProgressArea()
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/actions/WriteXMPAction.java]: Shouldn't this already on the FX thread by defintion?

Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/actions/WriteXMPAction.java]: No, `writeXMP` is invoked as a background task (but not properly implemented as `BackgroundTask`, leading to these ugly workaround - but I didnt have the motivation to implement it properly).

**Pull request merged**